samwise.js

```diff
@@ -3,27 +3,28 @@
  *
```

Comment:
SCRIPT-Consider extracting this repeated logic into a separate function to avoid duplication.